src/quantities.hxx

```diff
@@ -177,14 +166,220 @@ auto m_vis(auto &df, const std::string &outputname,
+        return met_3dvec.Dot(zeta) - (alpha * pzetaVis);
+    };
+    return df.Define(outputname, calculate_pzetamissvis, {p_1_p4, p_2_p4, met});
+}
+/**
+ * @brief function used to calculate mTdileptonMET, which is the transverse mass
+ * of the di-lepton system. The transverse mass is calculated using the
+ * vectoroperations::calculateMT function.
+ *
+ * @param df name of the dataframe
+ * @param outputname name of the new column containing the mTdileptonMET value
+ * @param p_1_p4 lorentz vector of the first particle
+ * @param p_2_p4 lorentz vector of the second particle
+ * @param met lorentz vector of the met
+ * @return a new dataframe with the new column
+ */
+auto mTdileptonMET(auto &df, const std::string &outputname,
+                   const std::string &p_1_p4, const std::string &p_2_p4,
+                   const std::string &met) {
+    auto calculate_mTdileptonMET = [](ROOT::Math::PtEtaPhiMVector &p_1_p4,
+                                      ROOT::Math::PtEtaPhiMVector &p_2_p4,
+                                      ROOT::Math::PtEtaPhiMVector &met) {
+        auto dileptonsystem = p_1_p4 + p_2_p4;
+        return vectoroperations::calculateMT(dileptonsystem, met);
+    };
+    return df.Define(outputname, calculate_mTdileptonMET,
+                     {p_1_p4, p_2_p4, met});
+}
+/**
+ * @brief function used to calculate the transverse mass of a particle. The
+ * transverse mass is calculated using the vectoroperations::calculateMT
+ * function.
+ *
+ * @param df name of the dataframe
+ * @param outputname name of the new column containing the mT value
+ * @param particle_p4 lorentz vector of the particle
+ * @param met lorentz vector of the met
+ * @return a new dataframe with the new column
+ */
+
+auto mT(auto &df, const std::string &outputname, const std::string &particle_p4,
+        const std::string &met) {
+    auto calculate_mt = [](ROOT::Math::PtEtaPhiMVector &particle_p4,
+                           ROOT::Math::PtEtaPhiMVector &met) {
+        return vectoroperations::calculateMT(particle_p4, met);
+    };
+    return df.Define(outputname, calculate_mt, {particle_p4, met});
+}
+
+/**
+ * @brief function used to calculate the pt of the dilepton + met system.
+ *
+ * @param df name of the dataframe
+ * @param outputname name of the new column containing the pt_tt value
+ * @param p_1_p4 lorentz vector of the first particle
+ * @param p_2_p4 lorentz vector of the second particle
+ * @param met lorentz vector of the met
+ * @return a new dataframe with the new column
+ */
+
+auto pt_tt(auto &df, const std::string &outputname, const std::string &p_1_p4,
+           const std::string &p_2_p4, const std::string &met) {
+    auto calculate_pt_tt = [](ROOT::Math::PtEtaPhiMVector &p_1_p4,
+                              ROOT::Math::PtEtaPhiMVector &p_2_p4,
+                              ROOT::Math::PtEtaPhiMVector &met) {
+        auto dileptonmet = p_1_p4 + p_2_p4 + met;
+        return (float)dileptonmet.Pt();
```

Comment:
could be done in one line

Reply:
This is the way i thought it would be best readable :)